import-scripts/validation_utils_py3.py

```diff
@@ -17,197 +22,243 @@
 
-    def run_validate_checks(self):
-        # TODO Add options to run specific checks
-        self.validate_sids_match_pids()
-
-
-class CDMFileValidator:
-    """Class to validate all CDM data. Currently, this class only validates the clinical sample file."""
-
-    def __init__(self, sample_file_path):
-        self.sample_file_validator = CDMSampleFileValidator(sample_file_path)
-
-    def validate_sample_file(self):
-        self.sample_file_validator.validate()
-
-    # TODO Define other file type validation functions here
-
-    def validate(self):
-        self.validate_sample_file()
+    def validate_study(self):
+        self.validate_sample_file_sids_match_pids()
 
+class AZValidator(ValidatorMixin):
+    """
+    Validates all AstraZeneca study data.
+    """
+    
+    def validate_study(self):
+        self.validate_gene_panels_present()
 
-if __name__ == "__main__":
+    def validate_gene_panels_present(self, gene_panel_dir=None):
+        """
+        Checks that the gene panels referenced in data_gene_matrix.txt are present in the gene panels directory.
+        """
+        if not gene_panel_dir:
+            gene_panel_dir = os.path.realpath(os.path.join(self.study_dir, "..", "gene_panels"))
+        
+        # Get unique list of referenced gene panels
+        df = self.load_file("data_gene_matrix.txt")
+        required_panels = set()
+        required_panels.update(df['mutations'])
+        required_panels.update(df['cna'])
+        required_panels.update(df['structural_variants'])
+        
+        # Get list of gene panels we actually have
+        actual_panels = self.load_gene_panel_ids(gene_panel_dir)
+        
+        if not required_panels.issubset(actual_panels):
+            missing_panels = required_panels - actual_panels
+            self.error(f"Could not find the required gene panels: {missing_panels}")
+    
+    def load_gene_panel_ids(self, gene_panel_dir):
+        stable_ids = []
+        
+        for basename in os.listdir(gene_panel_dir):
+            file = os.path.join(gene_panel_dir, basename)
+            if not os.path.isfile(file) or not re.match(r"data_gene_panel_.*\.txt", basename):
+                continue
+            with open(file, 'r') as fh:
+                first_line = fh.readline()
+            m = re.match(r"stable_id: (.*)", first_line)
+            if not m:
+                self.error(f"Could not parse stable id from gene panel file: {file}")
+                continue
+            stable_id = m.group(1).strip()
+            stable_ids.append(stable_id)
+        
+        if len(set(stable_ids)) != len(stable_ids):
+            self.warning("Found duplicate stable ids. Please check the gene panel files.")
+        return set(stable_ids)
+    
+def main():
+    # Setup logging
+    # TODO output logs as JSON for better Datadog integration? see: https://docs.datadoghq.com/logs/log_collection/python/
+    LOG_FILE = "/data/portal-cron/logs/validation_utils_py3.log"
```

Comment:
@callachennault I hard-coded this path, but if you have a better way to organize it feel free to change the code